### PR TITLE
feat(storybook): upgrade storybook and remove duplicate dependencies

### PR DESCRIPTION
**Pull request by virtuoushub**
Upgrade to https://github.com/storybookjs/storybook/releases/tag/v6.1.11

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Review by Rugvip — COMMENTED** [review]
Thanks! :grin:

Could you take a stab at deduplicating the `webpack` dependency and possible some other ones in `yarn.lock`? We're currently pulling in both webpack `4.44.1` and `4.44.2`, should stick to just the latter. Simplest way to do it is probably to delete the conflicting entries in `yarn.lock` and running `yarn install`.

**Comment by virtuoushub** [comment]
@Rugvip, thanks for the tip! Your suggestion worked in terms of removing `webpack@4.44.1`, however some other dependency versions bumped and are causing failures when [running `tsc`](https://github.com/backstage/backstage/pull/3583/checks?check_run_id=1519982836) (and [apparently `prettier`](https://github.com/backstage/backstage/pull/3583/checks?check_run_id=1519982791) 🤷🏻  ):

I hand edited `yarn.lock` to remove the offending `webpack@4.44.1` and that seemed to have worked. 

For future reference, is there another method of deduping to try that is less manual than hand editing, but more surgical than the `rm yarn.lock && yarn install` approach?

---


FWIW, fixing the `tsc` errors is non trivial.

I tracked down a [breaking change](https://github.com/helmetjs/helmet/blob/main/CHANGELOG.md#removed) in [Helmet](https://www.npmjs.com/package/helmet) that looks to [have a workaround](https://github.com/helmetjs/helmet/issues/235#issuecomment-674016883), but that would require a code change to a portion of the code this PR has nothing to do w/ so I want to avoid that if possible. "Pinning" the dependency to 4.0.0 via `"helmet": "~4.0.0",` as opposed to `"helmet": "^4.0.0",` solves that `tsc` error; but then there others that seem to relate to jest, which are painful to figure out what is happening, and not worth it just to dedupe IMHO.

**Comment by Rugvip** [comment]
> For future reference, is there another method of deduping to try that is less manual than hand editing, but more surgical than the `rm yarn.lock && yarn install` approach?

Yes, you can use https://github.com/atlassian/yarn-deduplicate! It will remove all duplicates though, so still a pretty broad change compared to removing select entries.

File: yarn.lock [hunk 1]
```diff
@@ -1063,7 +1635,87 @@
...
+    "@babel/plugin-transform-template-literals" "^7.12.1"
+    "@babel/plugin-transform-typeof-symbol" "^7.12.1"
+    "@babel/plugin-transform-unicode-escapes" "^7.12.1"
+    "@babel/plugin-transform-unicode-regex" "^7.12.1"
+    "@babel/preset-modules" "^0.1.3"
+    "@babel/types" "^7.12.7"
+    core-js-compat "^3.7.0"
+    semver "^5.5.0"
+
+"@babel/preset-env@^7.9.5":
```
Comment (Rugvip) [inline yarn.lock:1718]: Perhaps we can try deduping this one as well? I think it's the source of the majority of the duplicates that are left, and that the lockfile changes will look a lot saner afterwards

File: packages/backend-common/package.json [hunk 2]
```diff
@@ -70,7 +70,7 @@
     "@types/archiver": "^3.1.1",
     "@types/compression": "^1.7.0",
     "@types/concat-stream": "^1.6.0",
-    "@types/fs-extra": "^9.0.3",
+    "@types/fs-extra": "9.0.1",
```
Comment (virtuoushub) [inline packages/backend-common/package.json:73]: @Rugvip, is this change going to cause an issue?

Unfortunately, in order for me to get `tsc` to work, I needed to revert this and hard pin it back to `9.0.1`. See: f5908b983a0e8ff9e3337fda267a20474fb31baa for more details.

File: packages/create-app/templates/default-app/packages/backend/package.json.hbs [hunk 3]
```diff
@@ -27,7 +27,7 @@
     "@backstage/plugin-proxy-backend": "^{{version '@backstage/plugin-proxy-backend'}}",
     "@backstage/plugin-scaffolder-backend": "^{{version '@backstage/plugin-scaffolder-backend'}}",
     "@backstage/plugin-techdocs-backend": "^{{version '@backstage/plugin-techdocs-backend'}}",
-    "@octokit/rest": "^18.0.0",
+    "@octokit/rest": "18.0.6",
```
Comment (virtuoushub) [inline packages/create-app/templates/default-app/packages/backend/package.json.hbs:30]: @Rugvip, similar question to https://github.com/backstage/backstage/pull/3583#discussion_r539029858 just this time related to this commit ( ffa941bb0c2cfc1a5e3fc9fd55ceaf8e3e436636 ) and the resulting TypeScript / yarn.lock fun.

**Comment by virtuoushub** [comment]
I have updated this PR to include the [latest (as of this writing) Storybook release](https://github.com/storybookjs/storybook/releases/tag/v6.1.11).

**Comment by Rugvip** [comment]
@virtuoushub Sorry, didn't mean for you to run `yarn-deduplicate` here, we'd rather keep this change a bit more focused to make it easier to identity issues. Would be best to go back to one of the older commits without the broad deduplication 😅 

Regarding the other dependency changes, we'll likely want to avoid those. Better fix the actual type issues in code rather than pin to older versions. If those issues popped up from doing the broad deduplication I'd say skip them for now though, bumping storybook and webpack should most likely not lead to a TypeScript battle xP

**Review by Rugvip — APPROVED** [review]
Thanks, looks good!

**Comment by virtuoushub** [comment]
> It will remove all duplicates though, so still a pretty broad change compared to removing select entries.

FWIW there is an option to target individual packages via [progressive deduplication](https://github.com/atlassian/yarn-deduplicate#progressive-deduplication) (e.g. `npx yarn-deduplicate yarn.lock --packages webpack`).

**Comment by freben** [comment]
@virtuoushub seems some conflicting yarn.lock change got in before you - but this is ready to go after fixing that

**Comment by virtuoushub** [comment]
@freben / @Rugvip - `yarn.lock` conflict has been resolved.

**Pull request merged**